SharedPackages/BrowserServicesKit/Sources/Networking/Auth/OAuthClient.swift

```diff
@@ -101,6 +101,9 @@ public protocol OAuthClient {
     /// Use the TokenContainer provided
     func adopt(tokenContainer: TokenContainer)
 
+    // Creates a TokenContainer with the provided access token and refresh token, decodes them and returns the container
+    func decode(accessToken: String, refreshToken: String) async throws -> TokenContainer
```

Comment:
this is a new function used from the web message handler for `setAuthTokens` where we receive a accesstoken+refreshtoken from the FE and we need to decode it and adopt it